src/hints/state.rs

```diff
@@ -111,6 +110,9 @@ pub fn set_preimage_for_class_commitments(
         ap_tracking,
     )?;
 
+    log::debug!("Setting class trie mode");
+    exec_scopes.data[0].insert(vars::scopes::PATRICIA_TREE_MODE.to_string(), any_box!(PatriciaTreeMode::Class));
```

Comment:
Do you expect this to do The Right Thing with the nesting of exec scopes?

Reply:
Yes, the point is actually to store stuff like that in the root-level scope so that we don't need to worry about nesting.